plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -857,7 +857,7 @@ private void updateModelRegisterStateAsDone(
     void deployModelAfterRegistering(MLRegisterModelInput registerModelInput, String modelId) {
         String[] modelNodeIds = registerModelInput.getModelNodeIds();
         log.debug("start deploying model after registering, modelId: {} on nodes: {}", modelId, Arrays.toString(modelNodeIds));
-        MLDeployModelRequest request = new MLDeployModelRequest(modelId, modelNodeIds, false, true);
+        MLDeployModelRequest request = new MLDeployModelRequest(modelId, modelNodeIds, false, true, true);
```

Comment:
What if user define a flow template with deploy hidden model action in it ? Can user deploy any hidden model?

Reply:
Isn't hidden model management (register, deploy, undeploy, delete) is out of scope for AI-Flow?

Reply:
With a flow template, we will still consider it as user initiated right. So we will still check for access. Only if it is system initiated, we will bypass the access checks

Reply:
> Isn't hidden model management (register, deploy, undeploy, delete) is out of scope for AI-Flow?

Yes for normal process, but that's not a hard control. What if user just copy the hidden model id and define a flow template to deploy it?

Reply:
Can user see the hidden model id? I thought only superAdmin can see it.

Reply:
>Can user see the hidden model id? I thought only superAdmin can see it.

model id isn't hidden to users. But model content is hidden to users. During prediction, model id is being used. As user can make the prediction request, they can know the model id eventually.

Reply:
>Yes for normal process, but that's not a hard control. User can copy the hidden model id and define a flow template to deploy it.

We opened this [deploy request ](https://github.com/opensearch-project/ml-commons/blob/main/client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java#L260) to clients and then user initiated request flag is being set to true from code. So every request sent from ai-flow or any other clients should be treated as user initiated request too.